contract/models/account_analytic_account.py

```diff
@@ -183,7 +183,7 @@ def _prepare_invoice(self):
             'currency_id': currency.id,
             'journal_id': journal.id,
             'date_invoice': self.recurring_next_date,
-            'origin': self.name,
+            'origin': self.code or self.name,
```

Comment:
Code is stored as reference
https://github.com/OCA/contract/pull/122/files#diff-47b79491db78ebebcb372014ce9c7bf3L179

If you want change this field, Don't you should change both?